core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -55,29 +98,201 @@ func (l *logPollerWrapper) Ready() error {
+			return nil, nil, err
+		}
+
+		parsingContract, err := functions_coordinator.NewFunctionsCoordinator(coordinator, l.client)
+		if err != nil {
+			l.lggr.Error("LatestEvents: creating a contract instance for parsing failed")
+			return nil, nil, err
+		}
+
+		l.lggr.Debugw("LatestEvents: parsing logs", "nRequestLogs", len(requestLogs), "nResponseLogs", len(responseLogs), "coordinatorAddress", coordinator.Hex())
+		for _, log := range requestLogs {
+			gethLog := log.ToGethLog()
+			oracleRequest, err := parsingContract.ParseOracleRequest(gethLog)
+			if err != nil {
+				l.lggr.Errorw("LatestEvents: failed to parse a request log, skipping")
+				continue
+			}
+			resultsReq = append(resultsReq, evmRelayTypes.OracleRequest{
+				RequestId:           oracleRequest.RequestId,
+				RequestingContract:  oracleRequest.RequestingContract,
+				RequestInitiator:    oracleRequest.RequestInitiator,
+				SubscriptionId:      oracleRequest.SubscriptionId,
+				SubscriptionOwner:   oracleRequest.SubscriptionOwner,
+				Data:                oracleRequest.Data,
+				DataVersion:         oracleRequest.DataVersion,
+				Flags:               oracleRequest.Flags,
+				CallbackGasLimit:    oracleRequest.CallbackGasLimit,
+				TxHash:              oracleRequest.Raw.TxHash,
+				CoordinatorContract: coordinator,
+			})
+		}
+		for _, log := range responseLogs {
+			gethLog := log.ToGethLog()
+			oracleResponse, err := parsingContract.ParseOracleResponse(gethLog)
+			if err != nil {
+				l.lggr.Errorw("LatestEvents: failed to parse a response log, skipping")
+				continue
+			}
+			resultsResp = append(resultsResp, evmRelayTypes.OracleResponse{
+				RequestId: oracleResponse.RequestId,
+			})
+		}
+	}
+
+	l.lggr.Debugw("LatestEvents: done", "nRequestLogs", len(resultsReq), "nResponseLogs", len(resultsResp), "nextBlock", nextBlock, "latest", latest)
+	return resultsReq, resultsResp, nil
 }
 
 // "internal" method called only by EVM relayer components
 func (l *logPollerWrapper) SubscribeToUpdates(subscriberName string, subscriber evmRelayTypes.RouteUpdateSubscriber) {
-	if l.contractVersion == 0 {
+	if l.pluginConfig.ContractVersion == 0 {
 		// in V0, immediately set contract address to Oracle contract and never update again
-		err := subscriber.UpdateRoutes(l.routerContract, l.routerContract)
+		err := subscriber.UpdateRoutes(l.routerContract.Address(), l.routerContract.Address())
 		l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "subscriberName", subscriberName, "error", err)
-	} else if l.contractVersion == 1 {
+	} else if l.pluginConfig.ContractVersion == 1 {
 		l.mu.Lock()
 		defer l.mu.Unlock()
 		l.subscribers[subscriberName] = subscriber
 
-		// TODO remove when periodic updates are ready
-		_ = subscriber.UpdateRoutes(l.routerContract, l.routerContract)
+		// TODO remove when periodic updates are ready and integration tests are properly migrated
+		_ = subscriber.UpdateRoutes(l.routerContract.Address(), l.routerContract.Address())
```

Comment:
So our starting "coordinator" address is actually set to the router contract address?  This doesn't make sense to me.

Reply:
This line will disappear in the next PR, like the comment says. IT's only here so the current integration test can pass.